composer/trainer/trainer.py

```diff
@@ -163,20 +242,16 @@ class contains metrics relevant to the specific dataset. Set to ``None`` for no
             Ignored if ``torch_profiler_trace_dir`` and ``profiler_trace_file`` are not specified. (default: ``False``).
         torch_prof_with_flops (bool, optional): Estimate flops for operators.
             Ignored if ``torch_profiler_trace_dir`` and ``profiler_trace_file`` are not specified. (default: ``True``).
-        train_subset_num_batches (int, optional): If specified, finish every epoch early after training
-            on this many batches. This parameter has no effect if it is greater than ``len(train_dataloader)``.
-            If None (the default), then the entire dataloader will be iterated over.
-        eval_subset_num_batches (int, optional): If specified, evaluate on this many batches.
-            This parameter has no effect if it is greater than ``len(eval_dataloader)``.
-            If None (the default), then the entire dataloader will be iterated over.
-        deepspeed_config (Dict[str, Any], optional): Configuration for DeepSpeed, formatted as a dictionary
-            according to the `DeepSpeed documentation <https://www.deepspeed.ai/docs/config-json/>`_. If any
-            non-None value is provided, the trainer will initialize the DeepSpeed engine. (default: ``None``)
+        deepspeed_config (Dict[str, Any], optional): Configuration for DeepSpeed, formatted as a JSON
```

Comment:
Can you move the deep speed config above the profiling parameters?

Also, can we update the type annotation to be `Union[bool, dict]` and change the default value to `False`. It would be nice if you could set `deepspeed_config = True` to be equivalent to `deepspeed_config = {}`....it feels a tad bit weird that setting something to an empty dictionary has a very different effect than setting something to None.